### PR TITLE
feat: add deprecated subpaths

### LINKED ISSUE #3289: add 'deprecated' sub-paths for deprecated tabs
add 'deprecated' sub-paths for deprecated tabs in the documentation-framework

### PR DESCRIPTION
**Pull request by nicolethoen**
Closes #3289

### CODE REVIEW

File: packages/documentation-framework/components/example/example.js
Comment (jeffpuzzo) [inline packages/documentation-framework/components/example/example.js:109]: It doesn't look like `reactCoreDeprecatedModule` is defined, and I think we want to extend upon `react-table`'s module for this scope and not `react-core` (I assume we'll want both eventually, but for [this change](https://github.com/patternfly/patternfly-react/pull/8362), at least `react-table`).

Comment (nicolethoen) [reply packages/documentation-framework/components/example/example.js:109]: truee

File: packages/documentation-framework/routes.js
Comment (jeffpuzzo) [inline packages/documentation-framework/routes.js:54]: Do we still have plans to use `react-legacy` after this change and introduction of `react-deprecated`?

File: packages/documentation-framework/scripts/md/parseMD.js
Comment (jeffpuzzo) [inline packages/documentation-framework/scripts/md/parseMD.js:283]: If we keep this function as you've expanded upon it, we should update this comment.

File: packages/documentation-framework/scripts/md/parseMD.js
Comment (jeffpuzzo) [inline packages/documentation-framework/scripts/md/parseMD.js:284]: So as to not have to add new parameters to this function whenever a new condition arises, we might want to consider reworking this a bit. This is what I had in mind roughly, where if there's another "variant" at any point, we just have to update the 1 location in `getTsDocNameVariant`:

```
// Build unique names for components with a "variant" extension
function getTsDocName(name, variant) {
  return `${name}${variant ? `-${variant}` : ''}`;
}

function getTsDocNameVariant(source) {
  if (source.includes('next')) {
    return 'next';
  }

  if (source.includes('deprecated')) {
    return 'deprecated';
  }
}
```

and the 2 usages above would then look like the following:
`tsDocs[getTsDocName(name, getTsDocNameVariant(file))] = { name, description, props };`
and
`const name = getTsDocName(componentName, getTsDocNameVariant(source));`

File: packages/documentation-framework/routes.js
Comment (nicolethoen) [reply packages/documentation-framework/routes.js:54]: ultimately react-deprecated will replace the legacy here so I can remove it, but probably not until it's been removed after the breaking change. I don't want to remove it before then.
The deprecated work is going in the v5 branch

Comment (jeffpuzzo) [reply packages/documentation-framework/routes.js:54]: Got it, so maybe we make a follow-up issue to remove this when we're ready?

**Review by jeffpuzzo — APPROVED** [review]
Looks good. Once Jenny's PR is merged; https://github.com/patternfly/patternfly-react/pull/8362, this build should pass.

File: packages/v4/patternfly-docs/patternfly-docs.source.js
Comment (jeffpuzzo) [inline packages/v4/patternfly-docs/patternfly-docs.source.js:59]: Was this tested locally by any chance? This regex might conflict with the deprecated path regex on line 61. To prevent the potential for that we can remove the wildcard at the beginning of this line and replace it with `components` (e.g. `/components/TableComposable/...`).

Comment (nicolethoen) [reply packages/v4/patternfly-docs/patternfly-docs.source.js:59]: not recently but i'll do that

Comment (nicolethoen) [reply packages/v4/patternfly-docs/patternfly-docs.source.js:69]: @jeffpuzzo so i'm realizing now that this line will be better tested in the react-docs package where we have our own patternfly-docs.source.js file defined. patternfly-react does not consume org's v4 package, it's actually the other way around, so if these lines dont work in the react workspace when the change is made there, then we can update it here to match whatever worked in the workspace. If this is good to be merged, hopefully this will unblock the work to deprecate table

Comment (jeffpuzzo) [reply packages/v4/patternfly-docs/patternfly-docs.source.js:69]: Ah right, that makes sense.

File: packages/documentation-framework/scripts/webpack/webpack.base.config.js
Comment (jeffpuzzo) [inline packages/documentation-framework/scripts/webpack/webpack.base.config.js:124]: I think all of these path aliases can be removed now since we're no longer using package exports in react-core/table.

File: packages/v4/patternfly-docs/patternfly-docs.source.js
Comment (jeffpuzzo) [inline packages/v4/patternfly-docs/patternfly-docs.source.js:59]: Maybe for now, unless we want to add TODO comments explaining when we'd add this back in, we can remove this commented out line along with lines 43-44?

File: packages/documentation-framework/components/example/example.js
Comment (jeffpuzzo) [inline packages/documentation-framework/components/example/example.js:116]: I think we can use the condition `source === 'react-deprecated'` just once for the inclusion of both of these deprecated modules.

**Review by jeffpuzzo — APPROVED** [review]
Tested this locally against patternfly-react successfully. Code changes LGTM.

**Review by jenny-s51 — APPROVED** [review]
Looks great @nicolethoen !! 🚀

Also tested locally and I am able to now add demos under Table -> react-deprecated .

**Pull request merged**
